### PR TITLE
faster and const EntityTree::findByID()

### PR DESCRIPTION
**Pull request by AndrewMeadows**

### CODE REVIEW

File: libraries/entities/src/EntityTree.cpp [hunk 1]
```diff
@@ -913,18 +913,25 @@ void EntityTree::findEntities(RecurseOctreeOperation& elementFilter,
...
+    if (foundEntity && !foundEntity->getElement()) {
+        // special case to maintain legacy behavior:
+        // if the entity is in the map but not in the tree
+        // then pretend the entity doesn't exist
+        return EntityItemPointer(nullptr);
+    } else {
+        return foundEntity;
     }
-    return foundEntity;
 }
```
Comment (AndrewMeadows) [inline libraries/entities/src/EntityTree.cpp]: While looking at other stuff I noticed this circuitous logic in our code: when looking for an `EntityItem` by ID we would (1) find the item in the `_entityMap`, (2) get its `EntityTreeElement`, (3) walk entities in `EntityTreeElement` to find the entity again.

The new code will now: (1) find the `EntityItem` in the the `_entityMap` (2) return the entity if it has been added to any `EntityTreeElement`, else return `nullptr`.

The behavior hasn't changed but the path to get there is shorter.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/7159/app_12392_PR.apk

**Comment by AndrewMeadows** [comment]
**Test-plan**:

This needs a smoke test.  The `EntityScriptingInterface` relies heavily on `EntityTree::findEntityByID()`, so if scripts are working as expected and entities can be edited then this is still working.

**Comment by AndrewMeadows** [comment]
Meh, I'm going to merge this work into PR-12365 to reduce smoke-test overhead.

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/7159/HighFidelity-Beta-PR12392-38bc3120042a61e0a15a5e02a82f489af71b25df.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/7159/HighFidelity-Beta-PR12392-38bc3120042a61e0a15a5e02a82f489af71b25df.dmg

**Pull request closed without merging**